indico/modules/categories/client/js/components/CalendarLegend.jsx

```diff
@@ -0,0 +1,86 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Checkbox, Select} from 'semantic-ui-react';
+
+import {Translate} from 'indico/react/i18n';
+import './CalendarLegend.module.scss';
+
+function LegendItem({title, color, checked, url, isSpecial, onChange}) {
+  return (
+    <div styleName="legend-item">
+      <div styleName="color-square" style={{backgroundColor: color}} />
+      <span styleName={isSpecial ? 'italic' : undefined} style={{color: 'black'}}>
+        {url && !isSpecial ? <a href={url}>{title}</a> : title}
+      </span>
+      <Checkbox styleName="legend-checkbox" checked={checked} onChange={onChange} />
+    </div>
+  );
+}
+
+LegendItem.propTypes = {
+  title: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  url: PropTypes.string,
+  checked: PropTypes.bool,
+  isSpecial: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+};
+
+LegendItem.defaultProps = {
+  url: undefined,
+  checked: true,
+  isSpecial: false,
+};
+
+function CalendarLegend({items, groupBy, onFilterChanged, onElementSelected}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const parsedItems = items.map(({id, title, checked, url, color, textColor}, index) => (
+    <LegendItem
+      key={id}
+      title={title}
+      color={color}
+      textColor={textColor}
+      checked={checked}
+      onChange={(_, data) => onElementSelected(id, data.checked)}
+      url={url}
+      isSpecial={index === 0}
+    />
+  ));
+  const options = [
+    {text: Translate.string('Category'), value: 'category'},
+    {text: Translate.string('Location'), value: 'location'},
```

Comment:
I'd call this "Venue" since that's the term we use everywhere else as well? (same in the 'No location' text)

Reply:
Fixed